mrp_subcontracting_skip_no_negative/models/stock_move.py

```diff
@@ -0,0 +1,30 @@
+# Copyright 2023 Quartile Limited
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl)
+
+from odoo import models
+
+
+class StockMove(models.Model):
+    _inherit = "stock.move"
+
+    def _action_done(self, cancel_backorder=False):
+        moves_with_no_check = self.filtered(lambda x: x.is_subcontract).with_context(
+            skip_negative_qty_check=True
+        )
+        # For rather unlikely occassions where linked production is not in the right
+        # state.
+        for move in moves_with_no_check:
+            production_move = self.search([("move_dest_ids", "=", move.id)])
+            production = production_move.production_id
+            if production.reservation_state != "assigned":
```

Comment:
I don't understand what this has to be with no negative module. Why not simply pass the corresponding context?

Reply:
We added this for a specific scenario. Consider a component from a subcontracted BOM that requires 2 units. However, if we only have 1 unit available in our stock location and we transfer this single unit to the subcontracted location, then when we validate the subcontracting receipt, the quantity check will be ignored. Consequently, the MO will appear like below. We want to avoid this situation.

<img width="1439" alt="Screen Shot 2023-10-17 at 14 49 18" src="https://github.com/OCA/manufacture/assets/45355704/14836da1-4450-4ef2-acb0-7c93b4d9c5de">